src/main/java/gov/cabinetofice/gapuserservice/web/LoginController.java

```diff
@@ -64,4 +65,21 @@ public RedirectView redirectAfterColaLogin(final @CookieValue(name = REDIRECT_UR
 
         return new RedirectView(redirectUrl.orElse(configProperties.getDefaultRedirectUrl()));
     }
+
+    @GetMapping("/refresh-token")
+    public ResponseEntity<String> refreshToken(@CookieValue(USER_SERVICE_COOKIE_NAME) final String currentToken, final HttpServletResponse response) {
+
+        // presumably the security filter will handle invalid/non-existent tokens and deny access but if not then add code below.
```

Comment:
Can someone confirm/deny my thinking here pls

Reply:
Yep we'll just need to be careful to make sure the filters only running for the right endpoints